tools/ds-identify

```diff
@@ -1409,33 +1376,31 @@ dscheck_VMware() {
 
     # Check to see if running in a container and the VMware
     # datasource is configured via environment variables.
-    if vmware_has_envvar_vmx_guestinfo; then
-        if vmware_has_envvar_vmx_guestinfo_metadata || \
-            vmware_has_envvar_vmx_guestinfo_userdata || \
-            vmware_has_envvar_vmx_guestinfo_vendordata; then
+    if [ -n "${VMX_GUESTINFO}" ]; then
+        if [ -n "${VMX_GUESTINFO_METADATA}" ] || \
+           [ -n "${VMX_GUESTINFO_USERDATA}" ] || \
+           [ -n "${VMX_GUESTINFO_VENDORDATA}" ]; then
             return "${DS_FOUND}"
         fi
     fi
 
     # Do not proceed unless the detected platform is VMware.
-    if [ ! "${DI_VIRT}" = "vmware" ]; then
-        return "${DS_NOT_FOUND}"
-    fi
+    [ "${DI_VIRT}" = "vmware" ] || return ${DS_NOT_FOUND}
 
     # Do not proceed if the vmware-rpctool command is not present.
-    if ! vmware_has_rpctool; then
-        return "${DS_NOT_FOUND}"
-    fi
+    command -v vmware-rpctool >/dev/null 2>&1 || return ${DS_NOT_FOUND}
 
     # Activate the VMware datasource only if any of the fields used
     # by the datasource are present in the guestinfo table.
-    if { vmware_rpctool_guestinfo_metadata || \
-         vmware_rpctool_guestinfo_userdata || \
-         vmware_rpctool_guestinfo_vendordata; } >/dev/null 2>&1; then
-        return "${DS_FOUND}"
-    fi
+    local tmp=''
+    for tmp in "metadata" "userdata" "vendordata"; do
+      tmp="$(vmware-rpctool "info-get guestinfo.$tmp")"
```

Comment:
You could also rewrite this as:

```bash
local _key=""
local _val=""
for _key in "metadata" "userdata" "vendordata"; do
  if _val="$(vmware-rpctool "info-get guestinfo.${_key}")" && [ -n "${_val}" ]; then
    return "${DS_FOUND}"
  fi
done
```

Reply:
In fact, it could be simplified further like so:

```bash
local _key=""
local _val=""
for _key in "metadata" "userdata" "vendordata"; do
  if [[ -n "$(vmware-rpctool "info-get guestinfo.${_key}")" ]]; then
    return "${DS_FOUND}"
  fi
done
```

The above expression will fail if the command returns a non-zero exit code or empty string. Now, this requires double braces, which are not POSIX-shell compliant.

Reply:
rpctool returns a white space character, so `-n` will return 0
Also, for the tests to work I need to keep all the functions as it is. I will figure something out.

Reply:
>  I will figure something out.

Let's just leave the code as-is please. I'm not sure why you think it needs to change. Optimization for optimization's sake isn't always the right answer. I want to not require setting an env var in those tests as it makes them bound to serial execution.

As for rpctool, here you go :)

```bash
local _key=""
local _val=""
for _key in "metadata" "userdata" "vendordata"; do
  if _val="$(vmware-rpctool "info-get guestinfo.${_key}")" && [ -n "${_val//[:space:]]//}" ]; then
    return "${DS_FOUND}"
  fi
done
```

The use of `${_val//[:space:]]//}` strips the whitespace from the value.